builder/simplify.cc

```diff
@@ -1204,6 +1204,7 @@ class constant_bound : public node_mutator {
 
   template <typename T>
   void visit_logical(const T* op) {
```

Comment:
If this is now only used for equal/not_equal, maybe it'd be better to have more specific name for this function.